pkg/services/authn/authnimpl/registration.go

```diff
@@ -81,8 +81,20 @@ func ProvideRegistration(
 	}
 
 	if cfg.PasswordlessMagicLinkAuth.Enabled && features.IsEnabledGlobally(featuremgmt.FlagPasswordlessMagicLinkAuthentication) {
-		passwordless := clients.ProvidePasswordless(cfg, loginAttempts, userService, tempUserService, notificationService, cache)
-		authnSvc.RegisterClient(passwordless)
+		oauthInfos := socialService.GetOAuthInfoProviders()
+		enabledProvidersLen := 0
+		for _, provider := range oauthInfos {
+			if provider.Enabled {
+				enabledProvidersLen++
+			}
+		}
+
+		if cfg.SAMLAuthEnabled || enabledProvidersLen > 0 {
```

Comment:
It should also check whether LDAP is enabled.

Reply:
>  only users that have never used an external provider would be able to login with Passwordless (meaning only previously created users that have used username/password would be able to login with Passwordless)
That seems like the most straightforward solution. We only want basic auth users to be able to login via magic link.